io.mrarm.mcpelauncher-client.json

```diff
@@ -0,0 +1,113 @@
+{
+    "id": "io.mrarm.mcpelauncher-client",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.12",
+    "sdk": "org.kde.Sdk",
+    "command": "mcpelauncher-client",
+    "build-options" : {
+        "cflags": "-O2 -g -DNDEBUG",
+        "cxxflags": "-O2 -g -DNDEBUG",
+        "config-opts": [ "-DGAMEWINDOW_SYSTEM=GLFW", "-DCMAKE_THREAD_LIBS_INIT=-lpthread", "-DCMAKE_HAVE_THREADS_LIBRARY=1", "-DCMAKE_USE_PTHREADS_INIT=1" ]
+    },
+    "cleanup": [
+        "/include",
+        "/lib"
+    ],
+    "finish-args": [
+        "--socket=x11",
+        "--share=network",
+        "--filesystem=host",
+        "--share=ipc",
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--device=all",
+        "--talk-name=org.freedesktop.Flatpak"
+     ],
+    "modules": [
+        "pydeps.json",
+        {
+            "name": "msa-daemon",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir -p /app/bin/",
+                "echo '#!/bin/bash' > /app/bin/msa-daemon",
+                "echo 'flatpak-spawn --host flatpak --filesystem=~/.var/app/$FLATPAK_ID run app/io.mrarm.msa-daemon// \"$@\"' >> /app/bin/msa-daemon",
+                "chmod +x /app/bin/msa-daemon"
+            ]
+        },
+        {
+            "name": "mcpelauncher-client32",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/ChristopherHX/mcpelauncher-manifest.git",
+                    "branch": "flatpak",
+                    "disable-shallow-clone": true
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "cd mcpelauncher-linux-bin && git checkout armhf",
+                        "cd minecraft-symbols/tools && python3 ./process_headers.py --armhf"
+                    ],
+                    "only-arches": [ "arm" ]
+                }
+            ],
+            "buildsystem": "cmake-ninja",
+            "only-arches": [ "i386", "arm" ]
+        },
+        {
+            "name": "mcpelauncher-client",
+            "sources": [
+                {
```

Comment:
It's happy coincidence it works. SDK ships clang, libtinfo can be built from source as well with desired ABI version.

Reply:
Yes I know, SDK 5.14 ships clang-8, but it's one version too old. One new dependency of the 64bit compatible (experimental) launcher requires clang-9 or newer to compile sucessful.
Probably the upcoming freedesktop sdk 20.08 will have the right clang version in the sdk.
(libtinfo5 is a runtime dependency of the offical clang-9 release page, not referenced anywhere else)
> libtinfo can be built from source

done in my local version now.
So should clang be build from source too?

Reply:
SDK ships only clang-8, but the project needs **clang-9** or newer.
libtinfo is now build from source, but only a dependency of the standalone clang-9 compiler.
Newer sdk might solve this workaround.